packages/flutter/lib/src/material/two_level_list.dart

```diff
@@ -0,0 +1,146 @@
+
+const Duration _kExpand = const Duration(milliseconds: 200);
+
+class TwoLevelListItem extends StatelessComponent {
+  TwoLevelListItem({
+    Key key,
+    this.left,
+    this.center,
+    this.right,
+    this.onTap,
+    this.onLongPress
+  }) : super(key: key) {
+    assert(center != null);
+  }
+
+  final Widget left;
+  final Widget center;
+  final Widget right;
+  final GestureTapCallback onTap;
+  final GestureLongPressCallback onLongPress;
+
+  Widget build(BuildContext context) {
+    final TwoLevelList parentList = context.ancestorWidgetOfExactType(TwoLevelList);
+    assert(parentList != null);
+
+    return new SizedBox(
+      height: kListItemExtent[parentList.type],
+      child: new ListItem(
+        left: left,
+        center: center,
+        right: right,
+        onTap: onTap,
+        onLongPress: onLongPress
+      )
+    );
+  }
+}
+
+class TwoLevelSublist extends StatefulComponent {
+  TwoLevelSublist({ Key key, this.left, this.center, this.children }) : super(key: key);
+
+  final Widget left;
+  final Widget center;
+  final List<Widget> children;
+
+  _TwoLevelSublistState createState() => new _TwoLevelSublistState();
+}
+
+class _TwoLevelSublistState extends State<TwoLevelSublist> {
+  final AnimationController controller = new AnimationController(duration: _kExpand);
+  bool isExpanded = false;
+
+  void initState() {
+    super.initState();
+    isExpanded = PageStorage.of(context)?.readState(context) ?? false;
+    if (isExpanded)
+      controller.value = 1.0;
+  }
+
+  void handleOnTap() {
+    setState(() {
+      isExpanded = !isExpanded;
+      if (isExpanded)
+        controller.forward();
+      else
+        controller.reverse();
+      PageStorage.of(context)?.writeState(context, isExpanded);
+    });
+  }
+
+  Widget buildList(BuildContext context, Widget child) {
+    final ThemeData theme = Theme.of(context);
+
+    final easeOutAnimation = new CurvedAnimation(parent: controller, curve: Curves.easeOut);
+    final easeInAnimation = new CurvedAnimation(parent: controller, curve: Curves.easeIn);
+
+    final ColorTween borderColor = new ColorTween(begin: Colors.transparent, end: theme.dividerColor);
+    final Tween<double> heightFactor = new Tween<double>(begin: 0.0, end: 1.0);
+    final ColorTween headerColor = new ColorTween(begin: theme.text.body1.color, end: theme.accentColor);
+    final ColorTween iconColor = new ColorTween(begin: theme.unselectedColor, end: theme.accentColor);
+    final Animation<double> iconTurns = new Tween<double>(begin: 0.0, end: 0.5).animate(easeInAnimation);
+
+    return new Container(
+      decoration: new BoxDecoration(
+        border: new Border(
+          top: new BorderSide(color: borderColor.evaluate(easeOutAnimation)),
+          bottom: new BorderSide(color: borderColor.evaluate(easeOutAnimation))
+        )
+      ),
+      child: new Column(
+        children: <Widget>[
+          new TwoLevelListItem(
+            onTap: handleOnTap,
```

Comment:
s/easeInAnimation/_handleTap/

For consistency with other places in the framework.